src/ckb_chain/config.rs

```diff
@@ -0,0 +1,154 @@
+use ckb_types::{packed, prelude::*};
+use clap_serde_derive::ClapSerde;
+use molecule::prelude::*;
+use secp256k1::SecretKey;
+use std::{
+    io::{ErrorKind, Read},
+    path::PathBuf,
+};
+
+use crate::ckb::types::Hash256;
+
+pub const DEFAULT_CKB_CHAIN_BASE_DIR_NAME: &str = "ckb-chain";
+const DEFAULT_CKB_CHAIN_NODE_RPC_URL: &str = "http://127.0.0.1:8114";
+
+#[derive(ClapSerde, Debug, Clone)]
+pub struct CkbChainConfig {
+    /// ckb base directory
+    #[arg(
+        name = "CKB_CHAIN_BASE_DIR",
+        long = "ckb-chain-base-dir",
+        env,
+        help = format!("base directory for ckb chain actor [default: $BASE_DIR/{}]", DEFAULT_CKB_CHAIN_BASE_DIR_NAME)
+    )]
+    pub base_dir: Option<PathBuf>,
+
+    #[default(DEFAULT_CKB_CHAIN_NODE_RPC_URL.to_string())]
+    #[arg(
+        name = "CKB_CHAIN_NODE_RPC_URL",
+        long = "ckb-chain-node-rpc-url",
+        env,
+        help = "rpc url to connect the ckb node [default: http://127.0.0.1:8114]"
+    )]
+    pub rpc_url: String,
+
+    #[arg(skip)]
+    pub funding_source_lock_script_tx_hash: Hash256,
```

Comment:
Do you intend to create an new actor for every channel funding? User may want to open multiple channels to multiple users. It is my understanding that create an CkbChainActor to handle multiple channel opening would be better.

Reply:
This is for cell dep